tests/test_hierarchy_device.cu

```diff
@@ -431,20 +431,23 @@ BOOST_AUTO_TEST_CASE(amgx)
     boost::property_tree::info_parser::read_info("hierarchy_input.info",
                                                  *params);
     params->put("solver.type", "amgx");
-    params->put("solver.config_file", "amgx_config_amg.json");
+    params->put("solver.config_file", "amgx_config_fgmres.json");
 
     params->put("eigensolver.type", "lapack");
     params->put("agglomeration.nz", 2);
     params->put("laplace.n_refinements", 5);
     // We only supports Jacobi smoother on the device
     params->put("smoother.type", "Jacobi");
 
+    // TODO When using a finely refined mesh and two eigenvectors, AMGx does not
+    // converge. This needs to be invastigated. This can be due to a problem in
+    // AMGx configuration file, a problem when computing the eigenvectors with
+    // Lanczos or another bug.
+    params->put("eigensolver.number of eigenvectors", 1);
+
     // Relative tolerance in %
     double const tolerance_percent = 5.;
-    // The convergence rate for the two grid algorithm is 0.345914564 which is
-    // much better than the multigrid.
-    double const ref_solution = 0.86418797066393482;
-    double const conv_rate = test<dim>(params);
```

Comment:
I guess you didn't mean to remove this line.

Reply:
Fixed thanks